example.js

```diff
@@ -1,259 +1,127 @@
-        transitiveLayer._refresh()
-
-        let { paths, times } = data.paths
-
-        // they come out of r5 backwards
-        reverse(times)
-        reverse(paths)
-
-        // clear the ones that are the same and arrive at the same time
-        for (let p = 0; p < paths.length - 1; p++) {
-          // + 1: time is offset one minute (wait one minute and take the trip at the next minute)
-          if (times[p] === times[p + 1] + 1 && paths[p][0] === paths[p + 1][0] && paths[p][1] === paths[p + 1][1]) paths[p] = undefined
-        }
-
-        paths = await Promise.all(paths.filter(p => !!p).map(path => bc.getPath(path)))
-
-        // set up Marey plot
-        const marey = MareyFactory({dest: point, paths, times, transitiveData})
-        ReactDOM.render(marey, document.getElementById('marey'))
-
-        // and schematic line map
-        const lineMap = LineMapFactory({data: transitiveData})
-        ReactDOM.render(lineMap, document.getElementById('lineMap'))
-      } catch (e) {
-        console.error(e)
-      }
-    }
+/**
+ * Get a random point offset from the origin
+ */
+function generateDestinationLonLat ({
+  lat,
+  lon
+}) {
+  const latOffset = (getDistance() / LAT_DEGREE) * getSign()
+  const lonOffset = (getDistance() / (LAT_DEGREE * Math.cos(lat))) * getSign()
+  return {
+    lat: lat + latOffset,
+    lon: lon + lonOffset
   }
+}
 
-  // clickCount++ TODO: Get transitive working again
-})
-
-document.getElementById('show-isochrone').addEventListener('click', async function () {
-  // click again to hide
-  if (document.getElementById('isochrone').style.display === 'block') {
-    document.getElementById('isochrone').style.display = 'none'
-    return
-  }
-
-  document.getElementById('isochrone').style.display = 'block'
-
-  // figure out map pos
-  let bounds = map.getBounds()
-  let topLeft = bc.pixelToOriginPoint(map.project(bounds.getNorthWest()), map.getZoom())
-  let botRight = bc.pixelToOriginPoint(map.project(bounds.getSouthEast()), map.getZoom())
-
-  const contourGrid = ContourGridFactory({
-    contour: await bc.getContour(),
-    query: bc.query,
-    north: topLeft.y,
-    west: topLeft.x,
-    width: botRight.x - topLeft.x,
-    // unlike every other projection, south is larger in web mercator
-    height: botRight.y - topLeft.y
-  })
-
-  ReactDOM.render(contourGrid, document.getElementById('isochrone'))
-
-  return false
-})
-
-document.getElementById('isochrone-cutoff').addEventListener('input', function () {
-  updateIsoLayer()
-})
-
-document.getElementById('isochrone-play').addEventListener('click', function () {
-  const slider = document.getElementById('isochrone-cutoff')
-  let minute = 0
-
-  const interval = setInterval(function () {
-    slider.value = minute++
-    // trigger update
-    slider.dispatchEvent(new window.Event('input'))
-    if (minute > 120) clearInterval(interval)
-  })
+function getDistance () {
+  return Math.random() * MAX_DISTANCE + MIN_DISTANCE
```

Comment:
Only the minimum really mattered here. It's so that we don't need any click handling and can generate new destination values on page reload.